Aztec/Classes/Libxml2/DOMString.swift

```diff
@@ -462,28 +605,40 @@ extension Libxml2 {
         fileprivate func applyElement(_ elementName: String, spanning range: NSRange, equivalentElementNames: [String]) {
             
             let elementDescriptor = ElementNodeDescriptor(name: elementName, attributes: [], matchingNames: equivalentElementNames)
-            rootNode.wrapChildren(intersectingRange: range, inElement: elementDescriptor)
+            rootNode.wrapChildren(intersectingRange: range, inElement: elementDescriptor, undoManager: domUndoManager)
         }
         
         // MARK: - Candidates for removal
         
-        func removeLink(inRange range: NSRange){
-            domQueue.async { [weak self] in
-                self?.rootNode.unwrap(range: range, fromElementsNamed: ["a"])
+        func removeLink(inRange range: NSRange) {
+            performAsyncUndoable { [weak self] in
+                self?.removeLinkSynchronously(inRange: range)
             }
         }
         
         func updateImage(spanning ranges: [NSRange], url: URL, size: TextAttachment.Size, alignment: TextAttachment.Alignment) {
-            domQueue.async {
-                for range in ranges {
-                    let element = self.rootNode.lowestElementNodeWrapping(range)
+            performAsyncUndoable { [weak self] in
+                self?.updateImageSynchronously(spanning: ranges, url: url, size: size, alignment: alignment)
+            }
+        }
+        
+        // MARK: - Candidates for removal: Synchronously
+        
+        private func removeLinkSynchronously(inRange range: NSRange) {
+            rootNode.unwrap(range: range, fromElementsNamed: ["a"], undoManager: domUndoManager)
+        }
+        
+        private func updateImageSynchronously(spanning ranges: [NSRange], url: URL, size: TextAttachment.Size, alignment: TextAttachment.Alignment) {
+            
+            for range in ranges {
+                let element = self.rootNode.lowestElementNodeWrapping(range)
+                
+                if element.name == "img" {
```

Comment:
(Thought) we should have the Element Names stored somewhere. An enum / struct. For later!